### PR TITLE
improve the project-with-buildSrc exampe to show how to define extens…

### PR DESCRIPTION
**Pull request by jnizet**
…s and functions

Because it demonstrates a common use-case (as discussed in slack with Paul Merlin)

### Contributor Checklist
- [x] Base the PR against the `develop` branch
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Provide integration tests to verify changes from a user perspective
- [ ] Provide unit tests to verify logic
- [x] Ensure that tests pass locally: `./gradlew check --parallel`

### CODE REVIEW

File: samples/project-with-buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -1,2 +1,10 @@
 
 withHelloTask()
+
+tasks.create("printProfile") {
```
Comment (eskatos) [inline samples/project-with-buildSrc/build.gradle.kts]: :x: let's use `tasks.register()` here

File: samples/project-with-buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -1,2 +1,10 @@
...
+    description = "Uses the extension property and the extension functions defined in buildSrc. Use with -Pprofile=prod."
+    group = "sample"
+
+    println("The current profile is $profile")
```
Comment (eskatos) [inline samples/project-with-buildSrc/build.gradle.kts]: :x: This should be moved to a `doLast {}` block

File: subprojects/samples-tests/src/test/kotlin/org/gradle/kotlin/dsl/samples/ProjectWithBuildSrcSampleTest.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import org.hamcrest.MatcherAssert
+import org.junit.Test
+
+
+class ProjectWithBuildSrcSampleTest : AbstractSampleTest("project-with-buildSrc") {
+
+    @Test
+    fun `testProfile`() {
+        val output = build("printProfile").output
+        MatcherAssert.assertThat(
```
Comment (eskatos) [inline subprojects/samples-tests/src/test/kotlin/org/gradle/kotlin/dsl/samples/ProjectWithBuildSrcSampleTest.kt]: :x: please use static imports for `MatcherAssert` and `CoreMatchers` static members

File: samples/project-with-buildSrc/build.gradle.kts (hunk 2 above)
Comment (jnizet) [reply samples/project-with-buildSrc/build.gradle.kts]: Duh! Newbie mistake. I guess I'll never stop doing that one...

File: samples/project-with-buildSrc/build.gradle.kts (hunk 1 above)
Comment (jnizet) [reply samples/project-with-buildSrc/build.gradle.kts]: Sure.

File: samples/project-with-buildSrc/buildSrc/src/main/kotlin/extensions.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import org.gradle.api.*
+
+/**
+ * Defines an extension property of Project that can be used in the build script of the project (or any subproject)
+ */
+val Project.profile
+    get() = findProperty("profile") ?: "dev"
```
Comment (StefMa) [inline samples/project-with-buildSrc/buildSrc/src/main/kotlin/extensions.kt]: According to the [Kotlin Guidlines](https://kotlinlang.org/docs/reference/coding-conventions.html#property-formatting) this should be in one line

**Comment by eskatos** [comment]
Thank you @jnizet!

**Pull request merged**